src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -843,6 +853,15 @@ private static Task ProcessUnknownFrameAsync()
             {
                 throw new Http3StreamErrorException(CoreStrings.FormatHttp3DatagramStatusMismatch(_context.ClientPeerSettings.H3Datagram == 1, _context.ServerPeerSettings.H3Datagram == 1), Http3ErrorCode.SettingsError);
             }
+
+            if (string.Equals(HttpRequestHeaders.HeaderProtocol, WebTransportSession.WebTransportProtocolValue, StringComparison.Ordinal))
+            {
+                // if the client supports the same version of WebTransport as Kestrel, make this a WebTransport request
+                if (((AspNetCore.Http.IHeaderDictionary)HttpRequestHeaders).TryGetValue(WebTransportSession.CurrentSuppportedVersion, out var version) && string.Equals(version, "1", StringComparison.Ordinal))
```

Comment:
Make "1" a const on WebTransportSession? e.g. `WebTransportSession.Version1`

Reply:
A private variable, yes. Otherwise "1" is a magic string to anyone reading the code. A variable describes what the string is for.